src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -33,7 +34,7 @@ public class QueryInsightsSettings {
     /**
      * Max number of requests for the consumer to collect at one time
      */
-    public static final int QUERY_RECORD_QUEUE_CAPACITY = 1000;
+    public static final int QUERY_RECORD_QUEUE_CAPACITY = 100000;
```

Comment:
How is this decided? Please do benchmark tests with some real-world workload to justify the capacity limit.